src/Features/Core/Portable/QuickInfo/QuickInfoServiceWithProviders.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Extensions;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis.QuickInfo
+{
+    /// <summary>
+    /// Base class for <see cref="QuickInfoService"/>'s that delegate to <see cref="QuickInfoProvider"/>'s.
+    /// </summary>
+    internal abstract class QuickInfoServiceWithProviders : QuickInfoService
+    {
+        private readonly Workspace _workspace;
+        private readonly string _language;
+        private ImmutableArray<QuickInfoProvider> _providers;
+
+        protected QuickInfoServiceWithProviders(Workspace workspace, string language)
+        {
+            _workspace = workspace;
+            _language = language;
+        }
+
+        private ImmutableArray<QuickInfoProvider> GetProviders()
+        {
+            if (_providers == null)
```

Comment:
_providers is an ImmutableArray<> and therefore struct. Does this work?

Reply:
I think this changed at some point. Changed to `if (_providers.IsDefault)`.

Reply:
Looks like it was a copied and pasted from `CompletionServiceWithProviders`. Fixing there as well.